src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
The only reason the newFromArray methods exist is because DataValueDeserializer needs them. DataValue::getArrayValue serializes, newFromArray deserializes. That was the idea.

I don't care much about other code. What's relevant and really, really crucial here is to know how this is going to be used in DataValueDeserializer. I tried to add the contract of "newFromArray" including this big comment to the DataValue interface, but I figured this would make the situation worse. We already have an implementation without a "newFromArray", and we should be free to remove them individually when we are sure they are not used any more. That's why it should not be in the interface, and that's why the deprecation must mention the alternative.